### PR TITLE
Mustachio: generate renderer code for mixed in types

### PR DESCRIPTION
**Pull request by srawlins**
Also, regenerate renderers with `Package` visible. The Package class mixes in 4 classes which highlights the solution in this change; each of the 4 mixins gets its own renderer class.

Bringing in these major mixins brings the generated code to 2000 LOCs.

### CODE REVIEW

File: tool/mustachio/codegen_runtime_renderer.dart [hunk 1]
```diff
@@ -316,6 +341,26 @@ class ${renderer._rendererClassName}${renderer._typeParametersString}
         _buffer.writeln('    ...$superMapName$generics(),');
       }
     }
+    if (contextClass.mixins != null) {
+      // Mixins are spread into the property map _after_ the super class, so
+      // that they override any values which need to be overridden. Superclass
+      // and mixins override from left to right, as do spreads:
```
Comment (jcollins-g) [inline tool/mustachio/codegen_runtime_renderer.dart:347]: is this not right to left?  (N overrides M, which overrides E). Or am I thinking about this wrongly.

**Review by jcollins-g — APPROVED** [review]

File: tool/mustachio/codegen_runtime_renderer.dart (hunk 1 above)
Comment (srawlins) [reply tool/mustachio/codegen_runtime_renderer.dart:347]: Hmm yeah maybe the semantics are meaningless. I mean that mixins are _applied_ left to right. So M is applied, then N is applied, overriding things from M.

Comment (jcollins-g) [reply tool/mustachio/codegen_runtime_renderer.dart:347]: ah yes.  They are applied left to right, which implies that they override right to left.   Implied applies applies implying....

Comment (srawlins) [reply tool/mustachio/codegen_runtime_renderer.dart:347]: 😱 😱 😱

**Pull request merged**